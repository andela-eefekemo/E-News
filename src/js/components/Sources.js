import React from 'react';
import { Link } from 'react-router';

const Sources = (props) => {
  const { name, description, id } = props;
  return (
    <div className="col m4">
      <h5>{name}</h5>
      <p>{description}</p>
      <Link class="btn btn-default" to={`/headlines:&{id}`} >More Info</Link>
    </div>
  );
};
export default Sources;
