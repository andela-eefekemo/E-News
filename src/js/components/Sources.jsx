import React from 'react';
import { browserHistory } from 'react-router';
import PropTypes from 'prop-types';

// Sources Component
class Sources extends React.Component {

  handleQueryValue(href) {
    browserHistory.push(href);
  }
  render() {
    const { name, description, id, sortBysAvailable } = this.props;
    const link = `/headlines?name=${name}&id=${id}&sorts=${sortBysAvailable}`;
    return (
      <div className="col s12 m6 l4 ">
        <h5>{name}</h5>
        <p className="truncate">{description}</p>
        <button
          className="btn btn-default"
          onClick={this.handleQueryValue.bind(this, link)}>
          View Headlines
        </button>
      </div>
    );
  }
}

// Set default Props
Sources.defaultProps = {
  name: '',
  description: '',
  id: '',
  sortBysAvailable: [],
};

Sources.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  id: PropTypes.string,
  sortBysAvailable: PropTypes.array,
};

export default Sources;
