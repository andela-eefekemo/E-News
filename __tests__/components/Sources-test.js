/* global expect jest test */
import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Sources from '../../src/js/components/Sources';

describe('Sources Component', () => {
  test('should match the sources snapshot', () => {
    const component = mount(<Sources />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  test('should contain the appropraite props', () => {
    const wrapper = mount(<Sources />);
    expect(wrapper.node.props.name).toEqual('');
    wrapper.setProps({ name: 'the-next-web' });
    expect(wrapper.node.props.name).toEqual('the-next-web');
  });

  test('props should be in correct positions', () => {
    const wrapper = mount(<Sources name="Eguono" description="my name" />);
    const h5 = wrapper.find('h5');
    const p = wrapper.find('p');
    expect(h5.text()).toBe('Eguono');
    expect(p.text()).toBe('my name');
  });
});
