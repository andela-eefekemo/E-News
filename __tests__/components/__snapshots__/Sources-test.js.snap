// Jest Snapshot v1, https://goo.gl/fbAQLP

<<<<<<< HEAD:src/js/__tests__/components/__snapshots__/Sources-test.js.snap
exports[`Should Match the Sources Component 1`] = `
=======
exports[`Sources Component should match the sources snapshot 1`] = `
>>>>>>> b9dc9b0694863370ad929e16726c66fd2aa8f6a3:__tests__/components/__snapshots__/Sources-test.js.snap
<Sources
  description=""
  id=""
  name=""
  sortBysAvailable={Array []}
>
  <div
    className="col s12 m6 l4 "
  >
    <h5 />
    <p
      className="truncate"
    />
    <button
      className="btn btn-default"
      onClick={[Function]}
    >
      View Headlines
    </button>
  </div>
</Sources>
`;
